File: ReverseInteger/reverseInteger.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var res = 0;
    while(x) {
        res = res * 10 + x % 10;
        x = parseInt(x / 10);
    }
    res = (res > Math.pow(2,31) || -res > Math.pow(2,31)) ? 0 : res;
    return res;
};

File: ReverseInteger/reverseInteger.java
public class Solution {
    public int reverse(int x) {
        long res = 0;
        while(x != 0) {
            res = res * 10 + x % 10;
            x = x / 10;
        }
        res = (res < Integer.MIN_VALUE ||  res >  Integer.MAX_VALUE) ? 0 : res;
        return (int)res;
    }
}